### PR TITLE
Fix docs links

### PR DESCRIPTION
**Pull request by AlexisRenchon**
## Purpose 

Currently, the documentation of ClimaDiagnostics (in dev) has some broken links. This PR fixes those broken links.

Note: I followed documenters docs, specifically this section: https://documenter.juliadocs.org/stable/man/syntax/#@ref-link

### CODE REVIEW

**Comment by AlexisRenchon** [comment]
@Sbozzolo you can preview the docs here https://clima.github.io/ClimaDiagnostics.jl/previews/PR41/

I am pretty sure all links work now, but you can double check... I tried to write a better commit message too, as you suggested =)

**Comment by Sbozzolo** [comment]
Perfect!

If you add "Closes #8" to your PR message on GitHub, merging this PR will automatically close the issue.

File: docs/src/user_guide.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# What do I have to do to use `ClimaDiagnostics`?
+# [What do I have to do to use `ClimaDiagnostics`?](@id my_custom_header_name)
```
Comment (Sbozzolo) [inline docs/src/user_guide.md:1]: Maybe we want to use a different custom header name?

Comment (Sbozzolo) [reply docs/src/user_guide.md:1]: More descriprive

**Comment by AlexisRenchon** [comment]
Closes #8

**Pull request merged**